### ISSUE TITLE
PhpCAS attempts to validate the ticket more than once. 

### ISSUE THREAD

**Issue by nrbhatti** [Jasig/phpCAS#211]
PhpCAS attempts to validate the ticket more than once when there are more than one node. We are facing an interesting issue with our CAS setup. We have Zend-PHP web application using PhpCAS library. If we have 1 Zend Server running life is good, validation happens successfully. As soon as we add another node, application tries to validate the service ticket second time. Of course, authentication fails on second attempt. Looking at logs, it seems like first attempt was successful.  Have anybody faced similar issue? Maybe we are not doing something right ?

**Comment by jfritschi** [comment]
This sound like you have some issues with the sessions not being shared between your nodes? Phpcas needs the session to maintain the authentication status.

So either the session are shared between nodes or any client needs to be "sticky" to one node.

**Comment by nrbhatti** [comment]
Thanks for your response jfritschi. Session are being shared and we do not use sticky sessions on Zend servers. To learn more about the problem, we have bumped up number of max attempts a client can make to validate the service ticket (I know it should be 1). With 4 Zend servers, we set max attempts to 8. PhpClient now making 9 calls and now failing on 9th call. I am also seeing in Zend sc.log file this error 
Mod Cluster 31.08.2016 13:37:04 WARNING] [ mod_cluster.cpp : 1337 ( getSession ) ] Failed to get session. Reason: session ID is obsolete. 
[Mod Cluster 31.08.2016 13:37:04 NOTICE] [ mod_cluster.cpp : 621 ( ps_read_cluster ) ] failed to read session ST-86-ogXEPWId3En3Leq9GiKo

**Comment by jfritschi** [comment]
Have you looked at the debug log of phpcas?

**Comment by nrbhatti** [comment]
Yes, I have looked into that. I prints, couple of time user is already authenticated. No need for new ticket. Does not point to anything meaningful.

**Comment by nrbhatti** [comment]
I think I solved it. When initializing client, the last parameter is changeSessionId, the default value is true. Which I was using. The default value seems to be causing issue on our Zend Server with session clustering. If i set this value to false, the service ticket is being validated only once. This is method definition in php.cas file. We will deploy this next week. Thanks for your suggestions. 

/**
     * phpCAS client initializer.
     *
     * @param string $server_version  the version of the CAS server
     * @param string $server_hostname the hostname of the CAS server
     * @param string $server_port     the port the CAS server is running on
     * @param string $server_uri      the URI the CAS server is responding on
     * @param bool   $changeSessionID Allow phpCAS to change the session_id (Single
     * Sign Out/handleLogoutRequests is based on that change)
     */
    public static function client($server_version, $server_hostname,
        $server_port, $server_uri, $changeSessionID = true
    )

**Comment by jfritschi** [comment]
Good catch. This would have been my next suggestions.

Sometimes frameworks don't support that a session is renamed. 
Deactivating does however have one downside: You will have no single sign out capabilities for this application unless you create your own custom solution.
